packages/web-components/src/menu-item/menu-item.styles.ts

```diff
@@ -0,0 +1,194 @@
+import { css } from '@microsoft/fast-element';
+import { display } from '@microsoft/fast-foundation';
+import {
+  borderRadiusMedium,
+  colorCompoundBrandForeground1Hover,
+  colorCompoundBrandForeground1Pressed,
+  colorNeutralBackground1,
+  colorNeutralBackground1Hover,
+  colorNeutralBackground1Selected,
+  colorNeutralBackgroundDisabled,
+  colorNeutralForeground2,
+  colorNeutralForeground2Hover,
+  colorNeutralForeground2Pressed,
+  colorNeutralForeground3,
+  colorNeutralForegroundDisabled,
+  colorNeutralStrokeDisabled,
+  fontFamilyBase,
+  fontSizeBase200,
+  fontSizeBase300,
+  fontSizeBase500,
+  fontWeightRegular,
+  lineHeightBase200,
+  lineHeightBase300,
+} from '../theme/design-tokens.js';
+
+/** Menu styles
+ * @public
+ */
+export const styles = css`
+  ${display('grid')}
+
+  :host {
+    grid-template-columns: 20px 20px auto 20px;
+    align-items: center;
+    row-gap: 4px;
+    column-gap: 4px;
+    width: auto;
+    height: 32px;
+    background: ${colorNeutralBackground1};
+    font-family: ${fontFamilyBase};
+    font-size: ${fontSizeBase300};
+    font-weight: ${fontWeightRegular};
+    line-height: ${lineHeightBase300};
+    border-radius: ${borderRadiusMedium};
+    padding: 0 10px;
+    margin: 1px 0;
+    cursor: pointer;
+    overflow: visible;
+    contain: layout;
+  }
+  :host(:hover) {
+    position: relative;
+    z-index: 1;
+  }
+  .content {
+    white-space: nowrap;
+    flex-grow: 1;
+    padding: 0 2px;
+  }
+  .checkbox,
+  .radio {
+    display: none;
+  }
+  .expand-collapse-glyph-container,
+  ::slotted([slot='start']),
+  ::slotted([slot='end']),
+  :host([checked]) .checkbox,
+  :host([checked]) .radio,
+  .input-container {
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+  }
+  .content {
+    grid-column: auto / span 2;
+  }
+  .expand-collapse-glyph-container {
+    grid-column: 4 / span 1;
+    justify-self: flex-end;
+  }
+  .expand-collapse-glyph-container,
+  ::slotted([slot='start']),
+  ::slotted([slot='end']) {
+    width: fit-content;
+    height: 32px;
+    font-size: ${fontSizeBase500};
+  }
+  .input-container {
+    width: 20px;
+  }
+  ::slotted([slot='start']),
+  .expand-collapse-glyph-container,
+  .content,
+  .input-container {
+    color: ${colorNeutralForeground2};
+  }
+  ::slotted([slot='end']) {
+    width: fit-content;
+    color: ${colorNeutralForeground3};
+    font-size: ${fontSizeBase200};
+    font-weight: ${fontWeightRegular};
+    line-height: ${lineHeightBase200};
+    white-space: nowrap;
+    grid-column: 4 / span 1;
+    justify-self: flex-end;
+  }
+  :host(:hover) {
+    background: ${colorNeutralBackground1Hover};
+  }
+  :host(:hover) .expand-collapse-glyph-container,
+  :host(:hover) .content,
+  :host(:hover) .input-container {
+    color: ${colorNeutralForeground2Hover};
+  }
+  :host(:hover) ::slotted([slot='start']) {
+    color: ${colorCompoundBrandForeground1Hover};
+  }
```

Comment:
I want to be sure that I understand what this is doing...

For *anything* in the start slot the color of it should be `colorCompoundBrandForeground1Hover`. Is that accurate? Would this be different for the end slot?

Reply:
> Correct.. the start slot should only be uses for icons. So on hover the icons color is `colorCompoundBrandForeground1Hover` but this is different from the end slot which is used for "secondary content" which is just static text typically displaying info like short cut keys for that menu item action. Thoughts?
> 
> ![image](https://user-images.githubusercontent.com/11062709/221710417-d714c422-e570-4ef9-891c-8572ea6e7db8.png)

While the examples may not be exhaustive here, is there any reason someone couldn't add an icon in the end slot? Are we making assumptions that across every potential menu use the icon will always only ever be in the start position and/or no other type of slotted content visually would ever end up in the start slot? Should we consider mapping this directly to the `icon` attribute instance only? That would make this seem more reasonable IMO.

Reply:
@chrisdholt Are you suggesting that the hover state only exists when the `icon` attribute is present?

Reply:
Is there a scenario where the accent color should be applied to anything which is *not* an icon automatically? It seems like the design intent is for that to be applied to icons.

Reply:
I applied the `:hover` state to be dependent on the `icon` attribute.